cl/merkle_tree/hasher.go

```diff
@@ -15,10 +16,12 @@ const initialBufferSize = 0 // it is whatever
 // merkleHasher is used internally to provide shared buffer internally to the merkle_tree package.
 type merkleHasher struct {
 	// internalBuffer is the shared buffer we use for each operation
-	internalBuffer     [][32]byte
-	internalFlatBuffer []byte
+	internalBuffer           [][32]byte
+	internalFlatBuffer       []byte
+	internalBufferForSSZList [][32]byte
 	// mu is the lock to ensure thread safety
-	mu sync.Mutex
+	mu  sync.Mutex
+	mu2 sync.Mutex // lock onto ssz list buffer
```

Comment:
better to call it sszLock? or sszMu

Reply:
so mu2 is better because we can then use It in other places

Reply:
but ye i guess you right